### PR TITLE
Implement Task Details page 

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**Description of Changes**

- Refactored code to improve organization and reusability by consolidating state management and actions into the useTaskOperations hook
- Added TaskDetails component and page for viewing and editing individual tasks with detailed information
-  Slightly changed the appearance of the app 

https://github.com/user-attachments/assets/350268ea-4e82-4c7a-a35c-f0f0dd67264f

### CODE REVIEW

File: modules/HomePage/components/TaskList/Tasklist.module.css [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/* Assuming you are reusing TaskDetails.module.css, just add or adjust styles here if needed */
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/Tasklist.module.css]: Should remove the comments in all the files.

File: shared/utils/usetaskOperations.ts
Comment (ragib-hossain-sazim) [inline shared/utils/usetaskOperations.ts:1]: `usetaskOperations.ts` => `useTaskOperations.ts`

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 2]
```diff
@@ -104,12 +105,45 @@ const TaskList: React.FC<TaskListProps> = ({
...
         <Container
-          td={task.completed ? "line-through" : "none"}
           key={task.id}
           pt={10}
           mb="md"
-          pl={-10}
+          className={classes.taskContainer}
         >
+          <Link
+            style={{ textDecoration: "none", color: "inherit" }}
```
Comment (ragib-hossain-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx]: please don't inline styles. Move the styles in the corresponding styles folder.

File: modules/TaskDetailsPage/TaskDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            onChange={(value) => {
+              if (value) {
+                setEditFormState((prev) => ({
+                  ...prev,
+                  priority: parseInt(value, 10),
+                }));
+              }
+            }}
+            data={[
+              { value: "1", label: "High" },
```
Comment (ragib-hossain-sazim) [inline modules/TaskDetailsPage/TaskDetails.tsx]: define a constant for this options (for e.g 
```
export const PRIORITY_OPTIONS = [
{},
{},
{}
];
```

) in constants file and then use it here. Look into nextJS file/folder structure doc on where to define the constants file.

File: shared/utils/usetaskOperations.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        const nextState = prevRedoState[prevRedoState.length - 1];
+        setHistory((prevHistory) => [...prevHistory, tasks]);
+        setTasks(nextState);
+        return prevRedoState.slice(0, prevRedoState.length - 1);
+      }
+      return prevRedoState;
+    });
+  };
+
+  const applyFilter = (tasks: Task[], filter: string): Task[] => {
```
Comment (ragib-hossain-sazim) [inline shared/utils/usetaskOperations.ts:127]: This method is doing a lot of stuff together which makes it very difficult to read. Could you please simplify this method a bit. I suggest seperating 3 different filters in 3 different methods and then adjust the caller of this method according to the change.
for e.g: 

filterByStatus
filterByPriority
filterByDueDate

File: modules/HomePage/components/TaskList/Tasklist.module.css (hunk 1 above)
Comment (ragib-hossain-sazim) [inline modules/HomePage/components/TaskList/Tasklist.module.css]: Please remove the comments

**Comment by yamin-faruque-sazim** [comment]
Addressed the comments

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 5]
```diff
@@ -13,6 +14,8 @@ import {
 import { useDisclosure } from "@mantine/hooks";
 
 import { Task } from "../../types/Task";
+import classes from "./TaskList.module.css";
+import { PRIORITY_OPTIONS } from "@/shared/constants/taskContstants";
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx:18]: @ imports should be above relative imports

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**